macros/core/externalData.pl

```diff
@@ -0,0 +1,41 @@
+################################################################################
+# WeBWorK Online Homework Delivery System
+# Copyright &copy; 2000-2022 The WeBWorK Project, https://github.com/openwebwork
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of either: (a) the GNU General Public License as published by the
+# Free Software Foundation; either version 2, or (at your option) any later
+# version, or (b) the "Artistic License" which comes with this package.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See either the GNU General Public License or the
+# Artistic License for more details.
+################################################################################
+
+=head1 NAME
+
+externalData.pl - Macro that reads and writes data from a problem to the database.
+This is useful for using data common to multiple problems.
+
+=cut
+
+sub store_number_list {
+	my ($correct, $student, $ansHash, $value) = @_;
+	my @errors = ();    # stores error messages
```

Comment:
I just noticed this too.  Never define an array with `my @array = ();`.  It is just unnecessary.  Just use `my @array;`.  It is initialized to the empty array by default.  No need to explicitly add `= ()`.  The same goes for hashes.  Never do `my %hash = ()`.

Reply:
done.